src/__tests__/reduxForm.spec.js

```diff
@@ -100,6 +100,37 @@ const describeReduxForm = (name, structure, combineReducers, expect) => {
       }).toNotThrow()
     })
 
+    it('should update without error when there is no config', () => {
+      const store = makeStore()
+      const Form = () => <div />
+      const Decorated = reduxForm()(Form)
+
+      class Container extends Component {
+        constructor(props) {
+          super(props)
+          this.state = {}
+        }
+
+        render() {
+          return (
+            <Provider store={store}>
+              <div>
+                <Decorated form='formname' foo={this.state.foo} />
+                <button onClick={() => this.setState({ foo: 'bar' })} />
+              </div>
+            </Provider>
+          )
+        }
+      }
+
+      const dom = TestUtils.renderIntoDocument(<Container />)
+
+      expect(() => {
+        const button = TestUtils.findRenderedDOMComponentWithTag(dom, 'button')
+        TestUtils.Simulate.click(button)
+      }).toNotThrow()
+    })
+
```

Comment:
Nice! Thanks for adding a spec for this. The original change that broke this certainly seemed innocent, so this spec will help prevent similar regressions in the future.